OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -94,6 +95,12 @@ protected virtual Target RecalculateTarget(Actor self, out bool targetIsHiddenAc
 
 		public override bool Tick(Actor self)
 		{
+			if (!HaveArmamentsFor(target))
+				return true;
```

Comment:
we cannot just stop ticking child activity, we need to cancel it gracefully. Call `Cancel(self, true);` here instead of returning, and maybe don't call it if we are already canceling